core/schema/service.go

```diff
@@ -136,22 +145,64 @@ func (s *Service) Create(ctx context.Context, nsName string, schemaName string,
 func (s *Service) identifySchemaChange(ctx context.Context, request *changedetector.ChangeRequest) error {
 	endFunc := s.newrelic.StartGenericSegment(ctx, "Identify Schema Change")
 	defer endFunc()
+	schemaID, err := s.repo.GetSchemaID(ctx, request.NamespaceID, request.SchemaName)
+	if err != nil {
+		log.Printf("got error while getting schema ID from DB %s", err.Error())
+		return err
+	}
+	prevEvent, err := s.notificationEventRepo.GetByNameSpaceSchemaVersionAndSuccess(ctx, request.NamespaceID, schemaID, request.VersionID, true)
+	if err != nil && !errors.Is(err, pgx.ErrNoRows) {
+		log.Printf("got error while fetching previous notification event status for for namespace : %s, schema: %s, version: %d, %s", request.NamespaceID, request.SchemaName, request.Version, err.Error())
+		// todo push to grafana
+		return err
+	}
+	if prevEvent.ID != "" {
+		log.Printf("Duplicate request for schema change for namespace : %s, schema: %s, version: %d", request.NamespaceID, request.SchemaName, request.Version)
+		if _, err := s.notificationEventRepo.Update(ctx, prevEvent.ID, true); err != nil {
+			log.Printf("unable to update event for namesapce %s , schema %s and version %d in DB, got error: %s", request.NamespaceID, request.SchemaName, request.Version, err.Error())
+			return err
+		}
+		log.Printf("Update successful for schema change event")
+		return nil
+	}
 	sce, err := s.changeDetectorService.IdentifySchemaChange(ctx, request)
```

Comment:
I think the flow should be

First IdentifySchemaChange
Next Use that info (if needed) and create notificationEvent
and use the notification event (if needed) to push to kafka.

Reply:
its the same flow. Only some validation added.